tools/db_stress.cc

```diff
@@ -1794,6 +1794,7 @@ class StressTest {
           s = FLAGS_env->GetChildren(checkpoint_dir, &files);
         }
         DestroyDB(checkpoint_dir, Options());
+        delete checkpoint;
```

Comment:
This is safe only if memory allocation is successful and `checkpoint` points to valid memory space. Do we always have this assumption despite the current implementation of `Checkpoint::CreateCheckpoint`?

Reply:
@riversand963 would be better if `checkpoint` is initialize to nullptr and `Checkpoint::Create` only set it on success. If so, delete is alway safe even if `checkpoint == nullptr`. But I already click the land button..